### ISSUE TITLE
"Error! Resolution too small." at 320x240

### ISSUE THREAD

**Issue by OParczyk** [WyattBlue/auto-editor#142]
I'm using 09efd954f5c0de22ecb7a92efb26d81294473497.

**What went wrong**
I downloaded [https://www.youtube.com/watch?v=jNQXAC9IVRw](https://www.youtube.com/watch?v=jNQXAC9IVRw) using this command:
`youtube-dl -f bestvideo[ext=mp4]+bestaudio[ext=m4a] https://www.youtube.com/watch?v=jNQXAC9IVRw`
and ran `python3 -m auto_editor "Me at the zoo-jNQXAC9IVRw.mp4"`:

```
Error! Resolution too small.███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████] 100% done ETA 11:04 PM
```

**Does the example video work? (if applicable)**
Example video works.

**What does the program output when `--debug` is appended to your command/use case**

```
   - Temp Directory: /tmp/tmp7zznnhk7
   - Speeds: [99999, 1.0]                                                                                                                                                                                                                    
   - INPUT_FILE: Me at the zoo-jNQXAC9IVRw.mp4
   - newOutput: Me at the zoo-jNQXAC9IVRw_ALTERED.mp4
   - sampleRate: 44100
   - audioBitrate: 94k
Frame rate: 15.053003533568905
   - Frame Rate: 15.053003533568905
Analyzing audio volume.
  ⏳Creating new audio: [██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████] 100% done ETA 11:06 PM                
   - Total Samples: 837879
   - Samples per Frame: 2929.6478873239435
   - Expected video length: 285.99989904088847
Using opencv method
Error! Resolution too small.
   - Removed Temp Directory.
```
Notice, that the framerate is wrongly detected here too, I'll add a comment to #140.

**What's your platform? (if applicable)**

```
Python Version: 3.8.5 64-bit
Platform: Linux 5.4.0-65-generic
Config File path: /home/user/from_source/auto-editor/auto_editor/config.txt
FFmpeg path: ffmpeg
FFmpeg version: 4.2.4-1ubuntu0.1
Auto-Editor version 21w06a
```

**What's your input? (if applicable)**

```
file: Me at the zoo-jNQXAC9IVRw.mp4
 - fps: 15.053003533568905
 - duration: 18.866667
 - resolution: 320x240 (4:3)
 - video codec: av1
 - video bitrate: 136k
 - audio tracks: 1
   - Track #0
     - codec: aac
     - samplerate: 44.1 kHz
     - bitrate: 94k
```

**Comment by WyattBlue** [comment]
I couldn't replace the error using the same commands you did. It outputted just fine on both the av method and the opencv method.

I was able to verify the info auto-editor reported:

```
 - fps: 15.053003533568905
 - duration: 18.866667
 - resolution: 320x240 (4:3)
 - video codec: av1
 - video bitrate: 136k
 - audio tracks: 1
   - Track #0
     - codec: aac
     - samplerate: 44.1 kHz
     - bitrate: 94k
```

FFmpeg does report the fps being 15.05 on my machine though.

```
ffmpeg -hide_banner -i "Me at the zoo-jNQXAC9IVRw.mp4"
```

```
[libdav1d @ 0x7f88cd01c000] libdav1d 0.8.1
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'Me at the zoo-jNQXAC9IVRw.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2mp41
    encoder         : Lavf58.45.100
  Duration: 00:00:18.99, start: 0.000000, bitrate: 232 kb/s
    Stream #0:0(und): Video: av1 (Main) (av01 / 0x31307661), yuv420p(tv, bt470bg/unknown/unknown), 320x240, 135 kb/s, 15.05 fps, 15 tbr, 15360 tbn, 15360 tbc (default)
    Metadata:
      handler_name    : ISO Media file produced by Google Inc. Created on: 10/08/2018.
    Stream #0:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, mono, fltp, 93 kb/s (default)
    Metadata:
      handler_name    : SoundHandler
At least one output file must be specified
```

FFprobe tells a similar story. Running the command:

```
ffprobe -v error -select_streams v -show_entries stream=avg_frame_rate -of compact=p=0:nk=1 "Me at the zoo-jNQXAC9IVRw.mp4"
```

Returns

```
4260/283
```

and `4260 / 283` equals `15.053003533568905`

I'm curious what ffmpeg/ffprobe says about your video, but it appears the situation is either the following:
 1. FFmpeg/FFprobe is detecting the fps wrong.
 2. The program that you are using to get the video info is rounding down the fps.
 3. Our videos are different.

If the situation is indeed the 1st, this needs to be raised to the FFmpeg devs themselves.
2. 
I highly doubt 3 is the case but it's not impossible.

----

Going back on topic, what's stranger is that specific error, `Resolution too small`. That error is only designed to happen when either the width or the height of the video is less than 2 pixels. This information is captured by opencv/cv2.


```python
    # In auto_editor/renderVideo.py
    import cv2

    cap = cv2.VideoCapture(vidFile)
    width = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
    height = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
    fourcc = cv2.VideoWriter_fourcc(*'mp4v')

    if(args.scale != 1):
        width = int(width * args.scale)
        height = int(height * args.scale)

    if(width < 2 or height < 2):
        log.error('Resolution too small.')
```

So unless the args.scale variable is set close to 0 (connect to the `--scale` option, the default is set to 1), it should be impossible for that error to occur. Since it did occur however, I encourage you to poke around and see what's causing the bug to occur on your end.

For a quick stopgap, try using the av rendering method. `--render av`, that method doesn't raise this error.

**Comment by OParczyk** [comment]
Thank you for your investigation!
I just tried again using Python 3.6.12 instead of Python 3.8.5 and the problem didn't occur, so I digged deeper.
Turns out the packages supplied by Linux mint are somewhat old and opencv version 4.2.0 seems to have been the culprit.

Using all up to date packages (currently numpy 1.20.1, av 8.0.3 and opencv 4.5.1.48) installed using pip works in python 3.6 through 3.9.
I don't have the time to go through all the versions in between to determine the minimum working one, but since the installation of auto-editor using pip worked on my previously installed versions (numpy 1.17.4, opencv 4.2.0) I'd suggest re-evaluating the dependencies.
Maybe adding the version of dependencies to `auto-editor --debug` would also be worthwhile.

I tried my other issues to see if they persist using the updated modules and they do.
